xknx/devices/fan.py

```diff
@@ -93,23 +110,49 @@ def __init__(
 
     def _iter_remote_values(self) -> Iterator[RemoteValue[Any, Any]]:
         """Iterate the devices RemoteValue classes."""
-        yield from (self.speed, self.oscillation)
+        yield from (self.switch, self.speed, self.oscillation)
 
     @property
     def supports_oscillation(self) -> bool:
         """Return if fan supports oscillation."""
         return self.oscillation.initialized
 
-    async def set_speed(self, speed: int) -> None:
-        """Set the fan to a desginated speed."""
-        await self.speed.set(speed)
+    @property
+    def is_on(self) -> bool:
+        """Return the current fan state of the device."""
+        if self.switch.initialized:
+            return bool(self.switch.value)
+        return bool(self.current_speed)
+
+    async def turn_on(self, speed: int | None = None) -> None:
+        """Turn on fan."""
+        if self.switch.initialized:
+            await self.switch.on()
+            # For a switch GA fan, we only use an explicitly provided speed, but not
+            # arbitrarily set a default percentage here, compared to the speed GA based fans below.
+            await self.set_speed(speed)
+        else:
+            await self.set_speed(speed or DEFAULT_TURN_ON_PERCENTAGE)
+
+    async def turn_off(self) -> None:
+        """Turn off fan."""
+        if self.switch.initialized:
+            await self.switch.off()
+        else:
+            await self.set_speed(0)
+
+    async def set_speed(self, speed: int | None) -> None:
```

Comment:
Do we need to accept `None` here?
I think it would be better to guard in the calling methods to avoid an no-op-await.